src/unit_test.h

```diff
@@ -0,0 +1,111 @@
+/***********************************************************************
+ * Copyright (c) 2025  Matias Furszyfer (furszy)                       *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#ifndef SECP256K1_UNIT_TEST_H
+#define SECP256K1_UNIT_TEST_H
+
+/* --------------------------------------------------------- */
+/* Configurable constants                                    */
+/* --------------------------------------------------------- */
+
+/* Maximum number of command-line arguments.
+ * Must be at least as large as the total number of tests
+ * to allow specifying all tests individually. */
+#define MAX_ARGS 150
+/* Maximum number of parallel jobs */
+#define MAX_SUBPROCESSES 16
+
+/* --------------------------------------------------------- */
+/* Test Framework Registry Macros                            */
+/* --------------------------------------------------------- */
+
+#define CASE(name) { #name, run_##name }
+
+#define MAKE_TEST_MODULE(name) {\
+    #name, \
+    tests_##name, \
+    sizeof(tests_##name) / sizeof(tests_##name[0]) \
+}
+
+/* --------------------------------------------------------- */
+/* Test Framework API                                        */
+/* --------------------------------------------------------- */
+
+typedef void (*test_fn)(void);
+
+struct TestEntry {
+    const char* name;
+    test_fn func;
+};
+
+struct TestModule {
+    const char* name;
+    struct TestEntry* data;
+    int size;
+};
+
+typedef int (*setup_ctx_fn)(void);
+typedef int (*teardown_fn)(void);
+
+/* Reference to a test in the registry. Group index and test index */
+typedef struct {
+    int group;
+    int idx;
+} TestRef;
+
+/* --- Command-line args --- */
+struct Args {
+    /* 0 => sequential; 1..MAX_SUBPROCESSES => parallel workers */
+    int num_processes;
+    /* Specific RNG seed */
+    const char* custom_seed;
+};
+
+struct TestFramework {
+    /* Command-line args */
+    struct Args args;
+    /* Test modules registry */
+    const struct TestModule* registry_modules;
+    /* Num of modules */
+    int num_modules;
+    /* Registry for tests that require no RNG init */
+    const struct TestModule* registry_no_rng;
+    /* Specific context setup and teardown functions */
+    setup_ctx_fn fn_setup;
+    teardown_fn fn_teardown;
+};
+
+/* --------------------------------------------------------- */
+/* Public API                                                */
+/* --------------------------------------------------------- */
+
+/*
+ * Initialize the test framework.
+ *
+ * Must be called before tf_run() and as early as possible in the program.
+ * Parses command-line arguments and configures the framework context.
+ * The caller must set 'registry' and 'num_tests' before calling.
```

Comment:
```suggestion
 * The caller must set 'registry_modules' and 'num_modules' before calling.
```
(guess these fields origin from an earlier iteration of the PR and were then renamed at some point)